src/client/auth/Authenticator.java

```diff
@@ -0,0 +1,77 @@
+package client.auth;
+
+import client.controller.AdminController;
+import client.controller.ChefController;
+import client.controller.EmployeeController;
+import client.model.User;
+import client.requestgateway.AuthenticationRequestGateway;
+import client.util.JsonStringToObject;
+import client.util.RequestHandler;
+
+public class Authenticator {
+
+    private final AuthenticationRequestGateway requestGateway;
+    private final RequestHandler requestHandler;
+
+    public Authenticator() {
+        this.requestGateway = new AuthenticationRequestGateway();
+        this.requestHandler = new RequestHandler();
+    }
+
+    public boolean authenticate(String userId, String password) {
```

Comment:
The authentication method is also handling user roles. Please restructure your code to remove code that does not belong to authenticate function.